File: SuperMovie_server/src/api/Order/IsOrderIdValid.cs
namespace SuperMovie.Server.Api.Order;

using WebSocketSharp;
using WebSocketSharp.Server;

public struct IsOrderIdValidReq
{
    public long OrderId;
}

public struct IsOrderIdValidRsp
{
    public bool Ok;
}

//api : is_order_id_valid
public class IsOrderIdValid : WebSocketBehavior
{
}

File: movieui/src/scripts/ws/Order/isOrderIdValid.ts
import {wsUrlRoot} from "@/scripts/ws/meta"
import {recvMsg, reqStringify, rspParse, sendMsg} from "@/scripts/ws/helper"

export {isOrderIdValid}

const conn = new WebSocket(`${wsUrlRoot}/is_order_id_valid`)

type IsOrderIdValidReq =
    {
        OrderId: bigint
    }

type IsOrderIdValidRsp =
    {
        Ok: boolean
    }

async function isOrderIdValid(req: IsOrderIdValidReq) {

    const task = recvMsg(conn)
    sendMsg(conn, reqStringify(req)).then()
    const msg = await task

    return <IsOrderIdValidRsp>rspParse(msg)
}
